src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -647,13 +663,219 @@ const submitEncryptModeForm: ControllerHandler<
+) => {
+  const mapFieldIdToField = keyBy(formFields, '_id')
+  return responses
+    .filter((r): r is EmailResponse | MobileResponse => {
+      switch (r.fieldType) {
+        case BasicField.Email: {
+          const field = mapFieldIdToField[r._id]
+          if (!field || field.fieldType !== r.fieldType) return false
+          // Only filter out fields with auto reply set to true, or if field is verifiable.
+          return field.autoReplyOptions.hasAutoReply || field.isVerifiable
+        }
+        case BasicField.Mobile: {
+          const field = mapFieldIdToField[r._id]
+          if (!field || field.fieldType !== r.fieldType) return false
+          return field.isVerifiable
+        }
+        default:
+          return false
+      }
+    })
+    .map((r) => pick(r, ['fieldType', '_id', 'answer', 'signature']))
+}
+
+/**
+ * Encrypt submission content before saving to DB.
+ */
+const encryptSubmission: ControllerHandler<
+  { formId: string },
+  SubmissionResponseDto | SubmissionErrorDto,
+  ParsedStorageModeSubmissionBody
+> = async (req, res, next) => {
+  const { formId } = req.params
+
+  const logMeta = {
+    action: 'convertToEncryptMode',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  // Retrieve form definition. TODO: extract this step when validation is added.
+  const formResult = await FormService.retrieveFullFormById(formId)
+  if (formResult.isErr()) {
+    logger.warn({
+      message: 'Failed to retrieve form from database',
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  // Guardrail to prevent new endpoint from being used for regular storage mode forms.
+  // TODO (FRM-1232): remove this guardrail when encryption boundary is shifted.
+  if (!formResult.value.get('newEncryptionBoundary')) {
+    return res
+      .status(StatusCodes.FORBIDDEN)
+      .json({ message: 'This endpoint has not been enabled for this form.' })
+  }
+
+  // Retrieve public key.
+  const publicKey = formResult.value.publicKey
+
+  if (!publicKey) {
+    logger.warn({
+      message: 'Form does not have a public key',
+      meta: logMeta,
+    })
+    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Form does not have a public key',
+    })
+  }
+
+  const attachmentsMap: Record<string, Buffer> = {}
+
+  // Populate attachment map
+  req.body.responses.filter(isAttachmentResponse).forEach((response) => {
+    const fieldId = response._id
+    attachmentsMap[fieldId] = response.content
+  })
+
+  const encryptedAttachments =
+    await getEncryptedAttachmentsMapFromAttachmentsMap(
+      attachmentsMap,
+      publicKey,
+    )
+
+  const filteredResponses = filterSendableStorageModeResponses(
+    formResult.value.form_fields as unknown as FormFieldDto[],
+    req.body.responses.map((response) => {
+      if (isAttachmentResponse(response)) {
```

Comment:
Why do we need to strip out attachment content?

Reply:
I think we need to strip out the attachment content when getting `encryptedContent` but not here, since attachment fields will already be taken out by `filterSendableStorageModeResponses`. Will remove this in a bit!

Reply:
Done in [80d1cf3](https://github.com/opengovsg/FormSG/pull/6587/commits/80d1cf33c39d24f00488a4978d0b8e6e9d05609d), thanks!